plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Type.hs

```diff
@@ -80,6 +81,11 @@ data Term name uni fun ann
     -- This is the cutoff at which constructors won't get pointer tags
     -- See Note [Term constructor ordering and numbers]
     | Error !ann
+    -- TODO: worry about overflow, maybe use an Integer
+    -- TODO: try spine-strict list or strict list or vector
+    -- See Note [Constr tag type]
+    | Constr !ann !Word64 ![Term name uni fun ann]
```

Comment:
This should be a strict list I believe. Perhaps a custom one whose elements are always `Term`s. Gonna be a major pain in the ass though, since no `fmap`, `traverse` etc would work. Well, at least it should be a strict list.

Reply:
If we don't monomorphise it (and it's not clear that we gain anything from doing that, it's not like we can unpack Term) then we should still get `fmap` etc. There are many strict list types on Hackage even.

Reply:
TBH, I kind of want to fix this by having a "for-evaluation" term type anyway, then we can fix the name type etc which will actually let us unpack.